herddb-core/src/test/java/herddb/data/consistency/TableDataCheckSumTest.java

```diff
@@ -0,0 +1,119 @@
+import herddb.core.ReplicatedLogtestcase;
+import herddb.core.TestUtils;
+import herddb.mem.MemoryCommitLogManager;
+import herddb.mem.MemoryDataStorageManager;
+import herddb.mem.MemoryMetadataStorageManager;
+import herddb.model.ColumnTypes;
+import herddb.model.DataScanner;
+import herddb.model.Record;
+import herddb.model.StatementEvaluationContext;
+import herddb.model.Table;
+import herddb.model.TransactionContext;
+import herddb.model.commands.CreateTableSpaceStatement;
+import herddb.model.commands.CreateTableStatement;
+import herddb.model.commands.InsertStatement;
+import herddb.model.commands.TableConsistencyCheckStatement;
+import herddb.utils.Bytes;
+import herddb.utils.DataAccessor;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import org.junit.Test;
+
+
+/**
+ *
+ * @author Hamado.Dene
+ */
+public class TableDataCheckSumTest extends ReplicatedLogtestcase {
+
+    @Test
+    public void consistencyCheckReplicaTest() throws Exception {
+        final String tableName = "table1";
+        final String tableSpaceName = "t2";
+        try (DBManager manager1 = startDBManager("node1")) {
+
+            manager1.executeStatement(new CreateTableSpaceStatement(tableSpaceName, new HashSet<>(Arrays.asList("node1", "node2")), "node1", 2, 0, 0), StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
+            manager1.waitForTablespace(tableSpaceName, 10000, true);
+            try (DBManager manager2 = startDBManager("node2")) {
+                manager2.waitForTablespace(tableSpaceName, 10000, false);
+                manager1.executeStatement(new CreateTableStatement(Table.builder().tablespace(tableSpaceName).name(tableName).primaryKey("key").column("key", ColumnTypes.STRING).build()), StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
+                manager1.waitForTable(tableSpaceName, tableName, 10000, true);
+                manager1.executeStatement(new InsertStatement(tableSpaceName, tableName, new Record(Bytes.from_string("one"), Bytes.from_string("two"))), StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
+
+                manager1.executeStatement(new InsertStatement(tableSpaceName, tableName, new Record(Bytes.from_string("second"), Bytes.from_string("two"))), StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
+
+                manager2.waitForTable(tableSpaceName, tableName, 10000, false);
+
+                TableConsistencyCheckStatement statement = new TableConsistencyCheckStatement("table1", "t2");
+                manager1.createTableChekSum(statement, null);
+
+                manager1.executeStatement(new InsertStatement(tableSpaceName, tableName, new Record(Bytes.from_string("kk"), Bytes.from_string("kk"))), StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
```

Comment:
what is the meaning of this INSERT

This test looks incomplete.

here you are starting two servers 
you issue the TableConsistencyCheckStatement to the leader

what does node2 do ?
Most probably in this test you want to:
1) run only node1
2) request the check of table "t2"
3) get the result
4) validate the result